### PR TITLE
[IMP] sale_report: allow grouping by customer zipcode

### PR DESCRIPTION
**Pull request by ferdymercury**
Description of the issue/feature this PR addresses:
https://github.com/OCA/pos/issues/988

Current behavior before PR:
No way to analyze sales based on customer zip code

Desired behavior after PR is merged:
Zip code can be selected in the filter or group dropdown menus.

This PR supersedes https://github.com/odoo/odoo/pull/121594

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale/report/sale_report.py [hunk 1]
```diff
@@ -195,6 +197,7 @@ def _group_by_sale(self):
             p.product_tmpl_id,
             partner.country_id,
             partner.industry_id,
+            partner.zip
```
Comment (Feyensv) [inline addons/sale/report/sale_report.py:200]:
```suggestion
            partner.zip,
```
Missing `,` here (that's why the checks are red).

**Comment by Feyensv** [comment]
@robodoo r+

Removed the standard filters from the search view, they can be added manually by users when really needed, what matters the most is the ability to add those filters.

**Review by Feyensv — APPROVED** [review]
@robodoo r+

**Comment by Feyensv** [comment]
@robodoo rebase-ff

**Comment by fhe-odoo** [comment]
@Feyensv @dalaOdoo If we are going to allow group by zip, then nobody can argue against a group by State (which is still missing almost everywhere in Odoo) and a much more frequent need, especially in larger countries. In addition it's much less expensive since it's a m2o. Can you consider it?

**Comment by Feyensv** [comment]
@fhe-odoo Indeed, makes sense to support grouping by state as well :+1:
Soon fixed with #123691.

**Comment by ferdymercury** [comment]
Awesome, thanks so much for the implementation.

In the future, it might be very useful to support country- or zip-map view with colors proportional to sales, aka chorophlet maps, or with sized bubbles.

Such as https://youtu.be/NTD8-maahZg

**Pull request closed without merging**